### PR TITLE
Add jacoco support

### PR DESCRIPTION
**Pull request by lbk003**
Added jacoco support

The README has instructions on generating the report and viewing it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -101,8 +102,23 @@ The released version of the SDK can be downloaded from https://github.com/Appdyn
...
+    * Generate report
+    ```bash
+    cd <MY_SOURCE_FOLDER>/iot-java-sdk
+    ./gradlew test jacocoTestReport
+    ```
+    * View the report in your favorite browser
+    * For MacOS
+    ```bash
+    cd <MY_SOURCE_FOLDER>/iot-java-sdk
+    open ./sdk//build/jacocoHtml/index.html
```
Comment (mickey121) [inline README.md]: double slash

File: README.md [hunk 2]
```diff
@@ -101,8 +102,23 @@ The released version of the SDK can be downloaded from https://github.com/Appdyn
     * [C++ SDK and Sample App](https://github.com/Appdynamics/iot-cpp-sdk)
     * [Python Sample Code](https://github.com/Appdynamics/iot-rest-api-sample-apps)
 
+## Miscellaneous
+* Code Coverage 
+    * `jacoco` is integrated in the `build.gradle` file to track code coverage
```
Comment (zhouyaoji) [inline README.md]: The `build.gradle` file runs `jacoco` to track code coverage.

File: README.md [hunk 3]
```diff
@@ -101,8 +102,23 @@ The released version of the SDK can be downloaded from https://github.com/Appdyn
...
+    * Generate report
```
Comment (zhouyaoji) [inline README.md]: Here's how to generate the report:

Comment (zhouyaoji) [inline README.md]: 1. Generate the report with the following `gradlew` command:

File: README.md [hunk 4]
```diff
@@ -101,8 +102,23 @@ The released version of the SDK can be downloaded from https://github.com/Appdyn
...
+    ```bash
+    cd <MY_SOURCE_FOLDER>/iot-java-sdk
+    ./gradlew test jacocoTestReport
+    ```
+    * View the report in your favorite browser
```
Comment (zhouyaoji) [inline README.md]: 2. View the report by opening the

Comment (zhouyaoji) [inline README.md]: 2. View the report by opening the file `<MY_SOURCE_FOLDER>/iot-java-sdk/sdk/build/jacocoHtml/index.html` in a browser.

Comment (zhouyaoji) [inline README.md]: Delete this line.

File: README.md [hunk 5]
```diff
@@ -101,8 +102,23 @@ The released version of the SDK can be downloaded from https://github.com/Appdyn
...
+    * For MacOS
```
Comment (zhouyaoji) [inline README.md]: Delete this code snippet. We shouldn't have to provide special instructions for any OS. The same applies to all OS systems: open a file in a browser.

File: README.md (hunk 2 above)
Comment (zhouyaoji) [inline README.md]: Delete this line. It's not really a bullet item as much as information about how Gradle generates code coverage.

File: README.md [hunk 6]
```diff
@@ -101,8 +102,23 @@ The released version of the SDK can be downloaded from https://github.com/Appdyn
     * [C++ SDK and Sample App](https://github.com/Appdynamics/iot-cpp-sdk)
     * [Python Sample Code](https://github.com/Appdynamics/iot-rest-api-sample-apps)
 
+## Miscellaneous
```
Comment (zhouyaoji) [inline README.md]: ## Code Coverage 

It's not Miscellaneous because the section is specifically for code coverage.

File: README.md [hunk 7]
```diff
@@ -101,8 +102,23 @@ The released version of the SDK can be downloaded from https://github.com/Appdyn
...
+* Code Coverage
```
Comment (zhouyaoji) [inline README.md]: Delete this line as the section should be called "Code Coverage".

**Review by zhouyaoji — COMMENTED** [review]
I just made some suggestions for the **Miscellaneous** section.

File: README.md (hunk 7 above)
Comment (lbk003) [reply README.md]: I plan to add info about jacoco [Code Coverage], findbugs[Code Quality], checkstyle[Code Style] in this section hence I named it miscellaneous. @zhouyaoji 
Any other recommendations?

**Comment by mickey121** [comment]
LGTM

**Pull request merged**